src/Control/Monad.purs

```diff
@@ -28,9 +28,64 @@ import Data.Unit (Unit)
 -- | - Left Identity: `pure x >>= f = f x`
 -- | - Right Identity: `x >>= pure = x`
 -- | - Applicative Superclass: `apply = ap`
+-- |
+-- | ### Do Notation
+-- |
+-- | When using a type that has an instance for `Monad`, one can use
+-- | "do notation." In short, this code...
+-- | ```
+-- | foo =
+-- |   bind boxedA (\a ->
+-- |     let b = a + 4
+-- |     in bind boxedC (\c ->
+-- |          bind boxedUnit (\_ ->
+-- |            pure (a * b - c)
+-- |          )
+-- |     )
+-- |   )
+-- | ```
+-- |
+-- | ... can be converted into this code...
+-- | ```
+-- | foo =
+-- |   boxedA >>= (\a ->
+-- |     let b = a + 4
+-- |     in boxedC >>= (\c ->
+-- |          boxedUnit >>= (\_ ->
+-- |            pure (a * b - c)
+-- |          )
+-- |     )
+-- |   )
+-- | ```
+-- |
+-- | ...which can be converted once more into "do notation:"
+-- | ```
+-- | foo = do
+-- |   a <- boxedA
+-- |   let b = a + 4
+-- |   c <- boxedC
+-- |   boxedUnit
+-- |   pure (a * b - c)
+-- | ```
+-- |
+-- | Note: if one wants to use "do notation" but redefine what the in-scope
+-- | definitions are for `bind` and `pure` in a given context, one can use
+-- | "qualified do notation." This is an intermediate/advanced language feature
+-- | not explained here.
```

Comment:
I'm not sure that `Monad` is the best place to put documentation for do-notation. There isn't anything requiring a `Monad` instance in order to use it. This is a common Haskellism, since Haskell requires it, but if you already know to look for that, is this communicating new information?

Reply:
I thought it might be helpful for new users who are learning FP for the first time.

Reply:
I understand, and I don't think your intentions are misguided. I'm also not trying to say that the information isn't useful. My point is that core library documentation should strive for correctness. It is not correct to say that do-notation has anything to do with `Monad` (unlike Applicative-do, which certainly requires `Applicative`), even if we colloquially tie the two together. It's a common case that when you use do-notation, you likely also have a Monad instance, but this is not a constraint of the language or compiler. If I were a newcomer looking for information on do-notation, it is not clear to me that the `Monad` typeclass is documentation is the best place to put this information.

Reply:
> It's a common case that when you use do-notation, you likely also have a Monad instance, but this is not a constraint of the language or compiler.

Ah... because you can also do what I recall that you do with `do` (wow... that was a convaluted sentence :laughing: ):
```purescript
foo = do
 let x = 5 + 2
 5 * x
```
where "do notation" is a somewhat cleaner way of writing "let-in" blocks.

> If I were a newcomer looking for information on do-notation, it is not clear to me that the Monad typeclass documentation is the best place to put this information. It also is not clear to me that a newcomer is likely to encounter do-syntax for the first time by reading the documentation for Monad.

Good points! I thought having it here would be nice because one could find something like it in Pursuit when they search for things. However, I agree that there are other forms of documentation (e.g. the book, my repo) that cover it already and might be something new users consult first.

Reply:
I should also note that what I said was incorrect as well. Applicative-do does not "certainly" require `Applicative`, since it just inserts calls to `apply` and `pure` in scope, even if the common use is with the `Applicative` class.

Reply:
I'm fine with removing both of these do/ado notation docs.

Reply:
I've removed the do/ado notation explanations in latest commits.